_sips/minutes/2020-03-11-minutes.md

```diff
@@ -0,0 +1,220 @@
+    * Gui: some are common sense, some are for ambiguity, adding parameters (prioritisation)
+    * Martin: would be good to discuss the details with Miles
+    * Seb: shouldn't be in the "easy" list, then
+* Intersection/union
+    * Seb: no choice, and it's good
+* Opaque type aliases
+    * Seth: pushback from contributors, don't ship, wait for Valhalla
+    * Seb: they're not to replace value classes.  They have the runtime characteristics of type aliases.
+    * Lukas: can it be ported back to nsc?  Gui: no, sadly
+* Open classes
+    * Seb: summary: overall seems good, but details
+    * Seb: should it be scoped at the package? when should we warn/error?
+    * Gui: seems not easy
+    * Martin is against the counter-proposal of having it as an opt-in annotation
+* Parameter untupling
+    * Gui: the strong version is that anything on the left is a pattern (rather than just tuple)
+    * Seth: not crazy about it, seems like another thing you need to learn
+    * Gui: on the contrary, the status quo (having to add a `case`) forces you to learn this detail early
+    * Seb: but we're dropping auto-tupling?
+    * Martin: it's related to the pattern bindings changes (below)
+* Pattern bindings
+    * Martin `for (case Foo(a, b) <- ...)` for filtering, and val pattern, same
+    * https://dotty.epfl.ch/docs/reference/changed-features/pattern-bindings.html
+* Toplevel definitions
+    * Gui: details around `private`. top-level `private type` and `private class`, the type is in a synthetic object, the class is really top level
+    * that also impacts opaque types
+    * Seb: then let's force users to put in object and export instead
+    * but, boilerplate...
+* Trait parameters
+    * Gui/Seb: yes, everyone wants them
+    * Gui: something about mixing in multiple traits and them having implicit parameters (that need hand-wiring?)
+* Type lambdas
+    * Gui/Seb: yes, everyone wants them
+    * Gui: something about variance
+* Drop type projection (over abstract types)
+    * Gui: it's unsound
+    * Martin: Under `-strict` (i.e. how Scala 3.1 will behave by default) it's a compile error.  Will migrate from `-strict` to `-Xsource:3.1`
+    * Gui: we might change it so it's just lower-bonded, not upper-bounded
+    * Martin: actually, let's put that change under a `-Y` option
+    * Seb: Seems to me match types are the replacement for this
+* Weak conformance -> harmonization
+    * Seb: with overloading weak conformance is back
+    * Seb: Long and Any overloading, nsc conforms Int to Long, dotc picks Any
+    * Seb: JUnit assertEquals overloads, and Integer/Long aren't equals
+    * Seb: two parts to this change: the List case (inference) and overloading when primitives are present
+* Wildcard types
+    * `?` instead of `_`
+
+### Review enums
+
+Martin: most seem happy with the current design decisions
+
+two pushbacks:
+1. should apply return the precise type?
+2. support nested enums
+
```

Comment:
add: Martin says he's received a lot of feedback about the `apply` question that isn't reflected in the forum discussion. Gui said he'd post something.